src/server.c

```diff
@@ -3592,8 +3592,6 @@ void call(client *c, int flags) {
     /* Update failed command calls if required. */
 
     int is_failed = incrCommandStatsOnError(real_cmd, ERROR_COMMAND_FAILED);
-    is_failed = is_failed || c->deferred_reply_errors;
```

Comment:
The failure is introduced by this line of code

Reply:
i don't understand what you refer to. what failure, what caused it and what fixed it?

p.s. i'd rather named that var "failed", like we have "dirty".

Reply:
I changed the resulte of this condition by mistake so that some tests to fail.
from: 
```
if (!incrCommandStatsOnError(real_cmd, ERROR_COMMAND_FAILED) && c->deferred_reply_errors)
``` 
to:
```
int failed = incrCommandStatsOnError(real_cmd, ERROR_COMMAND_FAILED);
failed = is_failed || c->deferred_reply_errors;  // <-- causing result change
if (!failed && c->deferred_reply_errors)
```